storage/cloud.py

```diff
@@ -711,16 +715,23 @@ def __init__(
         endpoint_url=None,
     ):
         upload_params = {"ServerSideEncryption": "AES256"}
-        connect_kwargs = {}
+        client_kwargs = {
+            "config": Config(signature_version='s3v4')
+        }
+        resource_kwargs = {}
         if host or endpoint_url:
-            connect_kwargs["endpoint_url"] = endpoint_url or _build_endpoint_url(
+            client_kwargs["endpoint_url"] = endpoint_url or _build_endpoint_url(
+                host, port=port, is_secure=True
+            )
+            resource_kwargs["endpoint_url"] = endpoint_url or _build_endpoint_url(
```

Comment:
I don't think you don't need to specify two different params set. 
What's passed to the low-level client should also be passed to the Resource.
So I'd add the config to the existing _connect_params:
`connect_kwargs["config"] = Config(signature_version="s3v4"` instead.

Reply:
updated, thanks